### PR TITLE
adding a blog post for "percy magic"

### PR DESCRIPTION
**Pull request by mansona**
I wrote this Blog Post over the weekend when I was doing this work for the new Ember Website. I wasn't planning on continuing the "Emberification of the Ember Guides" series quite yet but sometimes things don't work out as you expect 😂 

### Todo

- [x] Figure out how we can represent a "series" in our blog (like a tag page that we can link to)
- [ ] Potentially port other parts in the series to the simplabs blog
  - https://blog.stonecircle.io/the-right-way-to-build-the-ember-guides/ (might need to be toned down a bit, especially the title 😂)
  - https://blog.stonecircle.io/broccoli-starting-the-build-and-eating-your-greens/

### CODE REVIEW

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
+---
+layout: article
+section: Blog
+title: "Porting a site to Ember using Percy"
+author: "Chris Manson"
+github-handle: mansona
+twitter-handle: real_ate
+topic: ember-guides
```
Comment (mansona) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I only put ember-guides as a topic because it was all I could think to do to make this a "series" 🤔 it's something that we should figure out

Comment (Turbo87) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I think this is meant to be more broad. In this case probably just `Ember`.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+This article will detail a technique where we used the visual testing service
+[Percy](https://percy.io/) to port the old [Ember
+Guides](https://guides.emberjs.com/release/) to use an entirely different
+technology without anybody noticing.
+
+![Screenshot from a talk Tom Dale gave saying "in some sense it's like trying to change an engine on a 747 mid-flight"](/images/posts/2019-03-02-porting-a-site-to-ember-with-percy/changing-the-engine.png)
+
+<!--break-->
+
+This article is part one of the series of posts where we describe how I ported the Ember
```
Comment (Turbo87) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: "we" vs. "I"?

Comment (jayjayjpg) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I like the idea to have this as an on-going series! Also I think it'd be

```suggestion
This article is part one of the series of posts where I describe how I ported the Ember
```

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+It's important to note that we weren't opposed to improving the end-user
+experience for this application. The whole reason for this project was that we
+wanted to make it **easier** to contribute improvements to the Ember Guides
+application, but we would release the technology re-write first without changing
+a single thing! This allowed us to have a straightforward requirement: **if it
+looks the same we can ship it.**
+
+## Enter Percy, stage right
+
+I had heard a lot about Percy over the years being a part of the Ember
```
Comment (Turbo87) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: maybe put another link here for people that forgot to click the first one ;)

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+If you are on a Linux machine you probably have `wget` installed, it is a
+particularly useful tool to download single files quickly and easily. If you are
+on macOS then you will probably need to use [Homebrew](https://brew.sh/) to
+install it:
+
+```bash
+brew install wget
+```
+
+Before I was working on this project I didn't actually realise that, not only
```
Comment (Turbo87) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]:
```suggestion
Before I was working on this project I didn't actually realize that, not only
```
😁

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+Percy installation instructions correctly, Percy will start to build your visual
+diff.
+
+The examples that I have been showing in this post have been leading up to [an
+actual PR on the ember-website
+project](https://github.com/ember-learn/ember-website/pull/185) and you will see
+in the CI checks for that PR that the [initial Percy build is
+failing](https://percy.io/Ember/Ember-Website/builds/1509496). If you look at
+the visual diff you will see that there is quite some work that needs to be done
+before this is ready to go live, but I never said that this would do all the work for you! 😂
```
Comment (jayjayjpg) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: lol

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+Here is what that code looks like in the ember-website:
+
+```javascript
+await pages.reduce(async (prev, config) => {
+  await prev;
+
+  await visit(config.route);
+
+  await percySnapshot(config.title);
+}, Promise.resolve());
```
Comment (Turbo87) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: since we're already using async/await here this can be simplified to a much more readable loop:
```js
for (let config of pages) {
  await visit(config.route);
  await percySnapshot(config.title);
}
```

**Review by Turbo87 — COMMENTED** [review]
nice work 👍

**Review by jayjayjpg — COMMENTED** [review]
This is a very detailed article and reads great! I just left one small suggestion and was also wondering if some of the things that are mentioned in the introduction paragraph could also find their way into a `Motivation` section which describes some of the benefits of porting the website over to an Ember app in more detail. In the contributor interviews about the deprecations app, some of the advantages that I remember were mentioned under the category "it makes it easier to contribute" were:

- familiarity of potential contributors with Ember (vs. Ruby)
- fast project setup on their local machine  (which seems to have been more difficult with the former middleman app

What do you think?

Otherwise this already seems :sparkles:

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 4 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: :trollface:

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 1 above)
Comment (wadie) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: Yes it should be `topic: ember` 👍

Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: so yea I get that we want this to be "topic: ember" but I'm wondering if we have the ability to have a "series" right now? 🤔 (or even if we want something like this 🙈)

My rough outline for this series originally was: 

- Strict requirements - Porting a site to Ember using Percy
  - this post 😂 
- Broccoli - Starting the Build and Eating Your Greens
  - this is already written elsewhere and we can port it over here
- The Build - the main process and the methodical work to get it working
- The Challenges - some of the issues we faced along the way
- Tech Highlights - prember, Fast Boot, Ember Data etc.
- The Launch - aka. The long tail - aka. the last mile

Comment (marcoow) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: This should just be `ember` for now - we can just have a block in the posts themselves that says each post is part of a series. I wouldn't recommend doing that before the second post though as the problem with series is usually that they stop after the first post ;)

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+twitter-handle: real_ate
+topic: ember-guides
+---
+
+This article will detail a technique where we used the visual testing service
+[Percy](https://percy.io/) to port the old [Ember
+Guides](https://guides.emberjs.com/release/) to use an entirely different
+technology without anybody noticing.
+
+![Screenshot from a talk Tom Dale gave saying "in some sense it's like trying to change an engine on a 747 mid-flight"](/images/posts/2019-03-02-porting-a-site-to-ember-with-percy/changing-the-engine.png)
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: This image seems unrelated to the text right before and after it?

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+
+<!--break-->
+
+This article is part one of the series of posts where we describe how I ported the Ember
+Guides from an old (and hard to maintain) ruby middleman app, to a modern Static
+Single Page Application that is a dream to contribute to! To see the rest of the
+posts and a description of the project you can take a look here (TODO - get a series going on the simplabs blog?).
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I think it't be good to add some more context here, e.g. the fact you're part of the Ember Learning team etc.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+
+## Strict Requirements
+
+The point of this project was to change the technology that powered the Ember
+Guides without altering the look or feel of the site when people were using it.
+There were many factors that we needed to consider, but this post is focusing on
+**just** the design, if you would like to see some of the other topics relating
+to the guides upgrade you can get out the series (TODO add link to series).
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: This seems similar to the previous paragraph - should those be merged maybe?

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+themselves as "Your all-in-one visual review platform". Just like Code Reviews
+and Pull requests, Percy allows you to view any **visual changes** as a result
+of the work that you are doing on an application. I have also heard it referred
+to as a "visual regression test system", but in practice it is much more than
+just that. 😂
+
+This got me thinking; if you can use Percy for visual regression tests
+then why can't you use it to do an overarching regression test for porting the
+technology that was powering your website, sort of like a "migration regression
+test". After a bit of experimentation that is what we decided to do.
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: Isn't this exactly what everyone means by _"visual regression testing"_ though? Making changes under the hood without any visual regressions?

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+
+## Setting up the baseline
+
+The first time that you run Percy it takes a snapshot of your entire site and
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]:
```suggestion
The first time that you run Percy it takes a snapshot of all pages on the site and
```

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+uses this as the "baseline" for which any subsequent change needs to be compared
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]:
```suggestion
uses these as the "baseline" for which any subsequent change needs to be compared
```

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 13]
```diff
@@ -0,0 +1,312 @@
...
+against.
+
+> It's not entirely true that Percy takes a snapshot of your entire site when
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: Is this paragraph intended as a comment for us? I think it's ok to say it takes screenshots of everything.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 14]
```diff
@@ -0,0 +1,312 @@
...
+what the site used to look like against the current changes.
+
+![An example of the Percy interface](/images/posts/2019-03-02-porting-a-site-to-ember-with-percy/percy-example.png)
+
+Although this article is part of the series to port the Ember Guides, we will
+use the project to convert the Ember Website to an Ember app for most of our
+examples. This is because, at the time of writing, we are going through much of
+the same process for the new Ember Website as we did when we originally
+converted the Ember Guides. The last step in the Ember Website conversion is to
+make sure it is 100% visually similar to the current website before we launch,
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: _"100% visually similar"_ sounds a bit strange tbh - I think either it's _"similar"_ only or 100% ;)

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 15]
```diff
@@ -0,0 +1,312 @@
...
+`wget` command. Most of these arguments are self-explanatory (and it could be an
+exercise for you to check the wget man page to see exactly what they do) but the
+most important one here is the `--reject-regex` for our use case. I mentioned
+already that the examples I'm giving for this article are for the
+"Emberificaion" of the Ember Website project, and currently for the website
+there are three sub-directories that are either currently or will soon be
+separate applications and are not something we need to concern ourselves within
+this project. This `--reject-regex` essentially excludes them from the download
+process, which is a good thing because the blog has 220+ pages that would all be
+automatically downloaded as part of this crawl without it 😬
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I'm not sure this whole section is needed to be honest - at least not with this level of detail. If you want to describe how Percy is great for changing the underlying implementation of something while making sure nothing changes visually (which I think is a good topic), I think this description of wget could be much shorter - just mentioning that you could use wget to obtain a copy of the page if you cannot run it locally is probably enough.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 16]
```diff
@@ -0,0 +1,312 @@
...
+./mascots?filter=other.html
+./mascots?filter=tomster.html
+./mascots?filter=corp.html
+./mascots?filter=meetup.html
+./mascots?filter=conference.html
+```
+
+If you have access to the output directory of the application that you're trying
+to migrate then you won't need to go through this step of cleaning up so I would
+recommend that if you can.
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: This section is maybe not necessary neither? It's pretty specific to the Ember Website after all and not really relevant for the readers I guess?

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 17]
```diff
@@ -0,0 +1,312 @@
...
+"make it look like it did before" and you can prevent feature creep for the
+project then you will know with confidence that it is ready to deploy when you
+have a passing CI and your Percy build shows no changes. If Percy is happy then
+you can deploy with confidence.
+
+## Summary
+
+This is a useful technique that we have now used successfully twice to port
+large projects from Ruby middleman to Ember in the Ember Learning Team, and
+while it is not exactly using Percy like it was designed it is a very useful
```
Comment (marcoow) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: > while it is not exactly using Percy like it was designed

Why not? Isn't this a perfect example for something like Percy actually?

**Review by marcoow — COMMENTED** [review]
This is a nice post! It illustrates very well a perfect use case for visual regression testing.

I'd consider changing the angle of the post slightly though - instead of _"we are moving the Ember Website from Ruby to Ember and here is how we used Percy for that"_, you could make that _"here is how you can use Percy to completely change how something is build while avoiding any visual regressions and here's an example of how we did all that while porting the Ember Website"_. I think that way the post might be more interesting to people who are interested in Percy but maybe not as much in the specifics of the Ember Website project.

You'll have to add your author image: 
bildschirmfoto 2019-03-07 um 16 03 51

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 10 above)
Comment (paytonealexis) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: To me this sentence could be interpreted 2 ways:
"...visual regression test for **building/maintaining** a website..." vs. migrating
or
That many people think that the most valuable part of visual regression testing is the moment it catches a regression, but in this case, you want to get to a state of 0 regressions/"visual changes". It seems obvious in this context why that's valuable, but it is often overlooked or misunderstood.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md [hunk 18]
```diff
@@ -0,0 +1,312 @@
...
+`wget` command. We need to provide the `PERCY_BRANCH` in our case because Percy
+is supposed to be run in conjunction with your Version Control System (VCS) and
+because we just have a folder instead of a git repository it won't autodetect
+our branch. We want it to be our `master` branch because this is going to serve
+as our baseline that all of the rest of the application will be built around.
+
+If you would like to see the result of this command you can follow [this link to
+the Percy build](https://percy.io/Ember/Ember-Website/builds/1509479). _Note
+this build will only be available for 90 days after this article goes live_.
+
```
Comment (paytonealexis) [inline blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: Sending you a note about your note now! :)

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 1 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I've changed it to just be `ember` and removed all mention of a series 👍

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 2 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: done 👍

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 3 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: done!

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 4 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: so... I don't know if we have a policy on this but I would always tend to write in "British English" 🤔

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 6 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: updated 👍

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 7 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I removed the image 👍

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 10 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: @marcoow this is intentionally different than "visual regression testing" because we're not trying to prevent changes in something that is done, we're seeding the system with essentially "fake" snapshots and using that as a bar to meet with our code

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 11 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: updated 👍

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 12 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: updated 👍

**Comment by mansona** [comment]
@marcoow I have updated this post quite considerably so it's probably worth another look.

I have an issue with your point about `I'd consider changing the angle of the post slightly though`, I understand that we would want to potentially target a wider audience but this post was mostly written as a historical record of something I actually did. I would suggest that if you would like me to write something for a wider audience then maybe we do a follow up blog with a different technology? 🤔 what do you think?

**Comment by mansona** [comment]
@jessica-jordan I really like your suggestion but the more I look at the article the less I feel like I know how to split out a motivation section 🙈 how much do you think it would benefit from having one? if it's worth it then maybe you could give me a hand with a 20 mins pairing session to help me with an outline and I can fill in the rest?

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 4 above)
Comment (marcoow) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: No policy as long as it is correct in either version of english.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 10 above)
Comment (marcoow) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: So you're always running all Percy builds against the same (static) reference?

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 19]
```diff
@@ -0,0 +1,288 @@
...
+bio: 'Senior Ember Devleoper'
+description: 'something something'
+og:
+  image: /assets/images/posts/2020-03-23-porting-a-site-to-ember-with-percy/og-image.png
+---
+
+This article details a technique where the visual testing software
+[Percy](https://percy.io/) was used to port the old [Ember
+Guides](https://guides.emberjs.com/release/) to an entirely different
+technology, without anybody noticing
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
technology, without anybody noticing.
```

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 20]
```diff
@@ -0,0 +1,288 @@
...
+
+<!--break-->
+
+This article details one of the first steps in the long road to _Emberify_ and
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: Can this paragraph start with slightly different wording? Having 2 subsequent paragraphs starting with _"This article details…"_ sounds a bit strange I think…

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 21]
```diff
@@ -0,0 +1,288 @@
...
+### Getting a copy of the existing site
+
+The first thing that we need in this process is a local copy of the existing
+site we are planning to port. If you look at the [documentation for using the
+Percy command line client](https://docs.percy.io/docs/command-line-client) it
+tells you that you should build your static site locally and then point the CLI
+client to the output folder.
+
+As I briefly mentioned earlier, part of the issue with the old infrastructure
+for the Ember Guides is that it was so hard to contribute to that it was almost
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
for the Ember Guides was that it was so hard to contribute to that it was almost
```

since it has been replaced now?

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 22]
```diff
@@ -0,0 +1,288 @@
...
+wget --recursive \
+     --page-requisites \
+     --html-extension \
+     --convert-links \
+     --domains emberjs.com \
+     emberjs.com
+```
+
+> I'm running these commands on macOS with `wget` version _1.19.5_ so you might
+> need to confirm these arguments for your platform and `wget` version
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
need to confirm these arguments for your platform and `wget` version.
```

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 23]
```diff
@@ -0,0 +1,288 @@
...
+```bash
+wget --recursive \
+     --page-requisites \
+     --html-extension \
+     --convert-links \
+     --domains emberjs.com \
+     emberjs.com
+```
+
+> I'm running these commands on macOS with `wget` version _1.19.5_ so you might
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
I'm running these commands on macOS with `wget` version _1.19.5_ so you might
```

We don't really have a styling for blockquotes currently. Also I think this isn't actually a quote anyway? :)

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 10 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: yes! the point about making sure that you create a branch is to ensure you don't reset the static reference 👍

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 24]
```diff
@@ -0,0 +1,288 @@
...
+```
+
+In this example emberjs.com is the name of the folder that was created with the
+`wget` command. We need to provide the `PERCY_BRANCH` in our case because Percy
+is supposed to be run in conjunction with your Version Control System (VCS) and
+since we just have a folder instead of a git repository it won't autodetect our
+branch. We want it to be our `master` branch because this is going to serve as
+the baseline around which all of the rest of the application will be built.
+
+## Setting up visual regression testing for the Ember Application
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: Is this a good heading when what you're doing is not really visual regression testing but more visual approximation or so?

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 13 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: Oh I was meaning to respond to this. It's intended to be an "asside" using the blockquote formatting, which it looks like we don't have any styles for 🤔 

You can see an example here: https://chris.manson.ie/the-true-power-of-json:api-have-someone-else-do-it#thetheory 

i.e. italics with some visual differentiation. Do we think that @ghislaineguerin could add a design for this in time? 🤔

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 14 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I removed this bit 🙈

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 15 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I added this section to make sure that people reading the article would have all the tools they needed to try out this slightly novel way of using Percy. I have simplified the section but I think the article still benefits from having this detail 🤔 what do you think?

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 25]
```diff
@@ -0,0 +1,288 @@
...
+powering your website, sort of like a "migration regression test". After a bit
+of experimentation that is what we decided to do.
+
+## Setting up the baseline
+
+The first time you run Percy it takes a snapshot of all pages on your site and
+uses these as the "baseline" for which any subsequent change needs to be
+compared against.
+
+> Percy doesn't always take a snapshot of your entire site; it depends on how
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
Percy doesn't always take a snapshot of your entire site; it depends on how
```

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 26]
```diff
@@ -0,0 +1,288 @@
...
+> you integrate their SDKs with your application. For the purpose of this article
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
you integrate their SDKs with your application. For the purpose of this article
```

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 27]
```diff
@@ -0,0 +1,288 @@
...
+> (and during the guides migration) we are using the "static site" command line
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
(and during the guides migration) we are using the "static site" command line
```

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 28]
```diff
@@ -0,0 +1,288 @@
...
+> interface to upload to Percy, which loops through all of your html files and
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
interface to upload to Percy, which loops through all of your html files and
```

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 29]
```diff
@@ -0,0 +1,288 @@
...
+> effectively uploads snapshots for all pages of your site.
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]:
```suggestion
effectively uploads snapshots for all pages of your site.
```

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 30]
```diff
@@ -0,0 +1,288 @@
+---
+title: 'Porting a site to Ember using Percy'
+author: 'Chris Manson'
+github: mansona
+twitter: real_ate
+topic: ember
+bio: 'Senior Ember Devleoper'
+description: 'something something'
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: This is still missing

**Review by marcoow — COMMENTED** [review]
This is a nice post – reads very well!

I'd add a bit more detail in the beginning about what's special about this. You could explain that while normally you'd have PRs with small changes that you compare against `master` step by step and merge then so they become part of the visual reference. I think what you were doing here is to collect all changes in a long-living branch and only once there were no visual differences between `master` (the reference you uploaded from the CLI) and the long-living branch, that branch could be merged? That is kind of clear but I'm not sure it'll be obvious to everyone. Even I'm not 100% sure I understood it correctly which is why I'm asking to confirm my understanding above ;)

**Comment by marcoow** [comment]
A general note about the different angles from which posts can be written and I think we should write posts: there is a substantial difference between posts that describe what **the author did** and posts that describe what **the reader can do** (potentially using something the author did as an example). The latter will generally be more interesting to a wider audience and I think we should generally prefer that angle. We can still write about the same things essentially, it's mainly a question of how we frame articles.

I understand rewriting this now would be quite some word – just something to consider for next time.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 10 above)
Comment (marcoow) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: See my review – this was not entirely clear to me and might not be for other readers. I'd add some more detail around that since that's really the key point of the entire post.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 13 above)
Comment (marcoow) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: Let's just make this a regular paragraph please. @ghislaineguerin has more important things to do currently.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 15 above)
Comment (marcoow) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: Yes, I think this makes sense in the latest version of the post.

File: blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md (hunk 13 above)
Comment (mansona) [reply blog/_posts/2019-03-02-porting-a-site-to-ember-with-percy.md]: I've removed the blockquotes for now 👍

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md (hunk 20 above)
Comment (mansona) [reply _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: I've updated it 👍 let me know if it needs any more work

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md (hunk 22 above)
Comment (mansona) [reply _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: I've applied all of these updates to remove the blockquotes, but pro tip: you can do multi line suggestions now 🎉 

> To add a comment on multiple lines, click and drag to select the range of lines, then click the blue comment icon.

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md (hunk 24 above)
Comment (mansona) [reply _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: Updated 👍

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md (hunk 30 above)
Comment (mansona) [reply _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: I've added it 👍 let me know if you would like any changes to it

File: _posts/2020-03-23-porting-a-site-to-ember-with-percy.md [hunk 31]
```diff
@@ -4,26 +4,31 @@ author: 'Chris Manson'
 github: mansona
 twitter: real_ate
 topic: ember
-bio: 'Senior Ember Devleoper'
-description: 'something something'
+bio: 'Senior Software Engineer, Ember Learning Core Team member'
+description: 'Chris Manson explains how to use Percy to port a site from one technology to another, while ensuring there are no visual changes.'
```
Comment (marcoow) [inline _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: I'd be more specific about that you're explaining how to use Percy to approximate a visual reference instead of ensuring you're not deviating from a moving reference. I think that's really the main point of the post and what makes it different from other guides to using Percy?

Comment (mansona) [reply _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: yes that's correct, that's for sure the main point but I am finding it hard to come up with a good one liner 🤔 

How about: 
```suggestion
description: 'Chris Manson details a novel way to use Percy to create a visual baseline that you can use to confidently port a site from one technology to another.'
```

Comment (marcoow) [reply _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: Or

```suggestion
description: 'Chris Manson details a novel way to use Percy for tracking approximation of a visual baseline when porting a site from one technology to another.'
```

?

Comment (mansona) [reply _posts/2020-03-23-porting-a-site-to-ember-with-percy.md]: got there with a bit of teamwork 🎉 😂

**Review by marcoow — APPROVED** [review]

**Comment by marcoow** [comment]
This is missing the `og-image` still and @ghislaineguerin might have feedback on the visual. Please @mansona and @ghislaineguerin sync so this can go out.

**Comment by ghislaineguerin** [comment]
@mansona @marcoow I will work on the visuals and og-image

**Comment by ghislaineguerin** [comment]
percy

**Comment by ghislaineguerin** [comment]
@mansona does this image work?

**Comment by mansona** [comment]
@ghislaineguerin I love it 😍 🎉

**Comment by mansona** [comment]
I've added the illustration @ghislaineguerin thanks again!

@marcoow I took a look at the og-image for your blog post, it seems to use a specific template. Do we have the source file somewhere so I can update the text for my one?

**Comment by marcoow** [comment]
@mansona easiest if @ghislaineguerin does it. We also need to 

- [x] update the date to today (which means also changing the name of the folder containing the images)

**Comment by mansona** [comment]
@marcoow what "old unused image" are you referring to? 🤔 

@ghislaineguerin if you create the og-image I can add it and update all the dates 👍

**Comment by marcoow** [comment]
ah, nevermind – I thought the new illustration replaced the previous image

**Review by marcoow — APPROVED** [review]
